runtime/pkg/ratelimit/ratelimiter.go

```diff
@@ -0,0 +1,143 @@
+package ratelimit
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-redis/redis_rate/v10"
+	"github.com/redis/go-redis/v9"
+	"github.com/rilldata/rill/runtime/pkg/observability"
+	"github.com/rilldata/rill/runtime/server/auth"
+	"math"
+	"net/http"
+	"reflect"
+)
+
+type RequestRateLimiter struct {
+	*redis_rate.Limiter
+}
+
+func NewRequestRateLimiter(redisAddr string) *RequestRateLimiter {
```

Comment:
As mentioned above, introduce an interface for a `Limiter` and then have implementations `ratelimit.RedisLimiter` and `ratelimit.Noop (that just allows all calls) initially.

Reply:
Thanks for the guide, looks better now